src/core/qgsogrutils.cpp

```diff
@@ -315,7 +323,7 @@ QVariant QgsOgrUtils::getOgrFeatureAttribute( OGRFeatureH ogrFet, const QgsField
   }
   else
   {
-    value = QVariant( QString() );
+    value = QVariant( field.type() );
```

Comment:
@elpaso Are you sure that the failing processing tests are unrelated, I didn't check thouroughly but they now seem to be consistant since this PR. Could this be caused by this change?

Reply:
That's possible, I'll have a look. In any case, I think that the patch is correct, the old behavior was to return an empty string even when the field type was not a string.

Reply:
@roya0045 I can confirm that this was indeed the problem. I'll find a solution.

Reply:
Thank you and apologies for the inconvenience. I agree that this fix is needed, though I'm not sure if the tests need to be updated or its something more that need to be modified.

Reply:
What apologies? Thank **you** for pointing me to the right direction!

I think that the issue is in the algorithms, they should check for NULL values before using the attribute values for calculations. The old behavior was relying on the fact that a NULL attribute was stored by OGR as an empty string instead of a NULL QVariant of the corresponding field type, the conversion of the `QVariant( QString())` to a numeric field was checked (and it was failing) so everything worked (at least for non-string types) but I think that it was a wrong implementation.